File: eidocell-backend/schemas/sessions.py
from datetime import datetime
from pydantic import BaseModel


class SessionCreate(BaseModel):
    name: str


class SessionUpdate(BaseModel):
    name: str | None = None
    scale_factor: float | None = None
    scale_units: str | None = None
    channel_names: list[str] | None = None


class SessionOut(BaseModel):
    id: str
    name: str
    images_directory: str | None = None
    session_folder: str
    created_at: datetime
    last_opened_at: datetime
    scale_factor: float
    scale_units: str
    channel_count: int
    channel_names: list[str] | None = None
    sample_count: int

    model_config = {"from_attributes": True}


class SessionListItem(BaseModel):
    id: str
    name: str
    images_directory: str | None = None
    created_at: datetime
    last_opened_at: datetime
    sample_count: int

    model_config = {"from_attributes": True}

File: eidocell-ui/src/types/sessions.ts
export interface SessionCreate {
  name: string
}

export interface SessionUpdate {
  name?: string
  scale_factor?: number
  scale_units?: string
  channel_names?: string[] | null
}

export interface SessionListItem {
  id: string
  name: string
  images_directory: string | null
  created_at: string
  last_opened_at: string
  sample_count: number
}

export interface SessionOut extends SessionListItem {
  session_folder: string
  scale_factor: number
  scale_units: string
  channel_count: number
  channel_names: string[] | null
}

export type PreviewPhase = 'ready' | 'importing' | 'pregenerating' | 'failed'

export interface PreviewStatus {
  ready: boolean
  progress: number
  message: string
  phase: PreviewPhase
}
